### PR TITLE
[inductor] Handle inductor counters in fx graph cache

### PR DESCRIPTION
**Pull request by masnesral**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130635

Summary: Similar to the handling of metrics, save inductor counter deltas in the FX graph cache entry and increment the counters appropriately on a cache hit

Test Plan: new unit test

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D59826029](https://our.internmc.facebook.com/intern/diff/D59826029)

### CODE REVIEW

**Review by eellison — APPROVED** [review]
nice!

**Comment by masnesral** [comment]
> AssertionError: "Simulate error" does not match "grad can be implicitly created only for scalar outputs"

I'm fairly certain this is unrelated to the change here

**Comment by masnesral** [comment]
@masnesral has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59826029).

File: test/inductor/test_codecache.py [hunk 1]
```diff
@@ -437,6 +437,39 @@ def fn(x, y):
...
+    @config.patch({"fx_graph_remote_cache": False})
+    def test_inductor_counters(self):
+        """
+        Test that we bump the inductor counters on a cache hit.
+        """
+
+        def fn(a, b):
+            return torch.mm(a, b)
+
+        a = torch.rand(8, 32, device="cuda")
```
Comment (etaf) [inline test/inductor/test_codecache.py:453]: If use requires_gpu, xpu will also run this case and failed on the cuda hard code. I suggest replace GPU_TYPE with "cuda", or replace requires_cuda with requires_gpu.

Comment (masnesral) [reply test/inductor/test_codecache.py:453]: Looks like you took care of this for me in https://github.com/pytorch/pytorch/pull/131006; thank you

**Pull request closed without merging**